### DISCUSSION TITLE
Saving to database locally but not production

### DISCUSSION THREAD

**Discussion by captainbooker** [pay-rails/pay#523] [Q&A]
When listening to stripe events locally everything saves to the database such as charges and subscriptions. 
However, when deeployed to heroku I don't recieve any errors in the stripe dashboard or heroku logs but it doesn't save anything after creating the customer. Doesn't save the charge nor subscription. Just saves the customer. 

Webhook endpoints are setup in the stripe dashboard and tested/fired events and you can see them come through the logs on heroku fine without error but not saving records for some reason...

**Comment by excid3** [comment]
Not sure, but you could fork and use `create!` here to have it raise an error if it can't create.

We should probably do that for all the webhook controllers.

**Comment by captainbooker** [comment]
@excid3 Well the Pay::Webhooks saves fine but after the webhook saves depending on what the event type is it should create a charge or subscription `Pay::Charge` or `Pay::Subscription ` which it doesn't in production.

After the webhook saves it does nothiing in production but works fine on local. I guess my question is where should we putting create! to raise an errror? 

This works fine and doesn't throw errors when creatiing the webhook. Just not sure why nothing saves afterwards
```
def create
        queue_event(verified_event)
        head :ok
      rescue ::Stripe::SignatureVerificationError => e
        log_error(e)
        head :bad_request
end
```

**Reply by excid3** [reply]
@dbookerkopitt Try from master now. I changed it to `create!` now so it will raise errors. cc45089
